Fix DET-curve doctest errors

dmohns opened the pull request:
I stumbled upon the 99% finished work in https://github.com/scikit-learn/scikit-learn/pull/8724 and thought we might as well finish it.

I haven't yet contributed to `scikit-learn` and couldn't not manage to get the doctest to run locally. But the problem looks pretty straigh-forward.

Commit:
Fix DET-curve doctest errors

- Sample snippet in  model_evaluation documentation was outdated.

jucor commented:
Thanks for picking it up, Daniel (@dmohns)! I'm on a somewhat poor connection at the moment, but will look at it when I'm back to civilization mid-next week.

dmohns commented:
Sure @jucor , whenever you have time. 

As side note I cannot get this branch to compile due to 
```
Error compiling Cython file:
------------------------------------------------------------
...
from libcpp.map cimport map as cpp_map

# Import the C-level symbols of numpy
cimport numpy as np

DTYPE = np.float64
         ^
------------------------------------------------------------

```
while it works flawlessly for `scikit-learn/master` branch. Could you rebase again to make development easier?

dmohns committed:
Clarify wording in DET-curve computation

- Align to the wording of ranking module to make it consistent.
- Add correct describtion of input and outputs.

dmohns commented:
As my attempts to rebase this went berserk I needed to close and re-open with cherry-picked commits. Please review carefully before merging.
